### PR TITLE
Chat improvements

### PR DESCRIPTION
**Pull request by JadaTBK**
Adds avatars, names, system messages.
Fixes hiding keyboard issues.

To Test
- join various activities
- send messages
- join as a different user & send messages
- leave activities to see system messages

To Do
- [ ] Blocked users / users removed from the chat
- [ ] Feedback regarding visuals / whatever else.

### CODE REVIEW

**Comment by JakeThurman** [comment]
Screen Shot 2019-04-03 at 9 31 00 PM

@gyurisic Do you think it's possible to make the system messages have no bubble (or a white/clear bubble)?

**Comment by JakeThurman** [comment]
I think we can just leave off the dates for system messages as well.

**Comment by JakeThurman** [comment]
I'm experiencing an issue where sometimes I can't end a message. Not sure what's causing it, but "Slip on a Rock" won't let me send any messages. I also see nothing in there which I feel like might also be wrong.

File: Buddies/Activities/View Activity/ViewActivityController.swift [hunk 1]
```diff
@@ -37,32 +37,22 @@ class ViewActivityController: UIViewController {
     
     /// Required for the `MessageInputBar` to be visible
     override var canBecomeFirstResponder: Bool {
-        
         if curMemberStatus! == .none {
```
Comment (JakeThurman) [inline Buddies/Activities/View Activity/ViewActivityController.swift]: Got a fatal error on this line. `curMemberStatus` was `nil`. You should probably have a default that assumes `.none`.

**Review by JakeThurman — CHANGES_REQUESTED** [review]
Screen Shot 2019-04-03 at 9 42 15 PM
Not sure if these are your fault, but in this screenshot^ I was able to send 2 messages before the system says I joined. The system message was there but my messages sent and were placed before it.

---

In this second screenshot you can see a related issue in setting these up, that they do not change when I change my name (in this case from "adds add" to "David McPerson Guy". (Again, don't think this is your fault but noting here.)

Screen Shot 2019-04-03 at 9 45 38 PM

File: Buddies/Activities/View Activity/ViewActivityController.swift (hunk 1 above)
Comment (noahtallen) [reply Buddies/Activities/View Activity/ViewActivityController.swift]: @JakeThurman I actually fixed this in some of my work. If we want that fix here:
```
let status = curMemberStatus ?? .none
if status == . none {}
```

**Comment by JadaTBK** [comment]
@JakeThurman I'll remove the dates and bubble from system messages. I'll look into how i handle member status which i think may be the cause of the non being able to end a message. I'll look into the weird ordering as well.

File: Buddies/Activities/View Activity/ViewActivityController.swift (hunk 1 above)
Comment (JadaTBK) [reply Buddies/Activities/View Activity/ViewActivityController.swift]: Fixed in the latest commit.

**Review by JakeThurman — APPROVED** [review]
g2g

**Pull request merged**